collector/asp_net_application.go

```diff
@@ -0,0 +1,160 @@
+package collector
+
+import (
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/yusufpapurcu/wmi"
+)
+
+func init() {
+	registerCollector("asp_net_application", newasp_net_applicationCollector) // TODO: Add any perflib dependencies here
+}
+
+// A asp_net_applicationCollector is a Prometheus collector for WMI Win32_PerfRawData_ASPNET_ASPNETApplications metrics
+type asp_net_applicationCollector struct {
+	ErrorsDuringExecution                *prometheus.Desc
+	ErrorsTotalPerSec                    *prometheus.Desc
+	ManagedMemoryUsedestimated           *prometheus.Desc
+	PercentManagedProcessorTimeestimated *prometheus.Desc
+	RequestsExecuting                    *prometheus.Desc
+	RequestsInApplicationQueue           *prometheus.Desc
+	RequestsPerSec                       *prometheus.Desc
+	RequestWaitTime                      *prometheus.Desc
+}
+
+func newasp_net_applicationCollector() (Collector, error) {
+	const subsystem = "asp_net_application"
+	return &asp_net_applicationCollector{
+		ErrorsDuringExecution: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "errors_during_execution"),
+			"The total number of errors that occur during the execution of an HTTP request, excluding parser and compilation errors.",
+			nil,
+			nil,
+		),
+		ErrorsTotalPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "errors_total_per_sec"),
+			"The number of errors per second that occur during the execution of HTTP requests, including any parser, compilation, or run-time errors.",
+			nil,
+			nil,
+		),
+		ManagedMemoryUsedestimated: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "managed_memory_used_estimated"),
+			"Estimated memory usage for the application",
+			nil,
+			nil,
+		),
+		PercentManagedProcessorTimeestimated: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "percent_managed_processor_time_estimated"),
+			"Estimated cpu usage for the application",
+			nil,
+			nil,
+		),
+		RequestsExecuting: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_executing"),
+			"The number of requests currently executing.",
+			nil,
+			nil,
+		),
+		RequestsInApplicationQueue: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_in_application_queue"),
+			"The number of requests currently in the queue.",
+			nil,
+			nil,
+		),
+		RequestsPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_per_sec"),
+			"The number of requests executed per second.",
+			nil,
+			nil,
+		),
```

Comment:
Similar situation here: if it's a counter metric it'll need the `_total` suffix and an updated metric description.